### ISSUE TITLE
Rule Proposal: Forbid naming variables beginning with "new"

### ISSUE THREAD

**Issue by MrHen** [sindresorhus/eslint-plugin-unicorn#251]
Forbid variables being named with `new` because it looks like the `new` keyword. `newFoo` and `new Foo()` are fairly similar when scanning code quickly (especially without syntax highlighting) and `newFoo` is already a pretty terrible variable name.

Fail:

* `const newFoo = 'foo';`

Pass:

* `const fooNew = 'foo';`
* `const foo = new Foo();`

----

A similar argument could be made for `class` and a few other keywords but it seems like `newFoo` is the most common scenario I've been bumping into recently.

**Comment by sindresorhus** [comment]
While I’ve never seen this naming used, it definitely makes sense to prevent it.

**Comment by futpib** [comment]
I don't think it makes sence, I've literally just wrote this and I don't think it could be written better:

```js
        if (newLocation !== location) {
            push(newLocation);
        }
```

**Comment by sindresorhus** [comment]
@futpib That's also a good point. I guess I'm neutral on this now.

**Comment by MrHen** [comment]
I want to explicitly forbid that use case. `new` has special meaning in JavaScript and having it at the beginning of a variable makes it look just enough like `new Location` that my brain has to stop and think about it when quickly scanning code. I've stopped using it in my code and find it _much_ more comfortable.

As for an alternative, you can just move `new` to the end and it fixes the issue because `new Location` and `locationNew` look totally different.

It's super nitpicky, I agree, but I was going to write a rule for this anyway and figured I could submit it to unicorn instead of making a standalone rule.
